### PR TITLE
Avatar Improvements

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- `AvatarCombo`, `AvatarIcon` & `AvatarUser` now supports `role="button"`
  - Added support for common DOM properties and event handlers (e.g.: `onClick`)
  - Improved a11y for all `Avatar*` components
  - Added Storybook with knobs

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [ ] Update documentation
- [x] Updated CHANGELOG
- [x] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] PR is ideally < ~400LOC

### CODE REVIEW

**Comment by lukelooker** [comment]
> Looks great, the only thing I would suggest is to add this to the documentation?

Great point! Added :)

File: packages/components/src/Avatar/AvatarUser.tsx [hunk 1]
```diff
@@ -38,24 +39,33 @@ export interface AvatarUserProps extends AvatarProps {
...
+    <BaseElement {...omitStyledProps(props)} aria-label={name}>
+      <AvatarInitials color={color} aria-hidden>
         {size === 'xxsmall'
           ? `${firstInitial}`
           : `${firstInitial}${lastInitial}`}
       </AvatarInitials>
       {user && user.avatar_url && (
-        <AvatarPhoto color={color} type="image/png" data={user.avatar_url} />
+        <AvatarPhoto
+          aria-hidden
```
Comment (jhardy) [inline packages/components/src/Avatar/AvatarUser.tsx:62]: This should be `aria-hidden="true"` does react do that for us?

Comment (jhardy) [reply packages/components/src/Avatar/AvatarUser.tsx:62]: Same on line 55 for `<AvatarInitials color={color} aria-hidden>`

Comment (lukelooker) [reply packages/components/src/Avatar/AvatarUser.tsx:62]: Yep. (I had to double-check myself) 

image

**Pull request merged**